library/common-library/templates/_hostnetwork.tpl

```diff
@@ -7,20 +7,41 @@ Returns "true" if `hostNetwork` is enabled, otherwise fallbacks to a overridable
 {{- /* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists */ -}}
 {{- $global := index .Values "global" | default dict -}}
 
-{{- /* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */ -}}
+{{- /*
+`get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs
+
+We also want only to return when this is true, returning `false` here will template "false" (string) when doing
+an `(include "common.hostNetwork" .)`, which is not an "empty string" so it is `true` if it is used
+as an evaluation somewhere else.
+*/ -}}
 {{- if get .Values "hostNetwork" | kindIs "bool" -}}
-    {{- .Values.hostNetwork -}}
+    {{- if .Values.hostNetwork -}}
+        {{- .Values.hostNetwork -}}
+    {{- end -}}
 {{- else if get $global "hostNetwork" | kindIs "bool" -}}
-    {{- $global.hostNetwork -}}
+    {{- if $global.hostNetwork -}}
+        {{- $global.hostNetwork -}}
+    {{- end -}}
 {{- else -}}
     {{- include "common.hostNetwork.defaultOverride" . -}}
```

Comment:
why defaultOverride? isn't take into consideration merely if nothing else is set?

Reply:
The name in the `common-library` itself could be misleading but when you use the library as it is called "defaultOverride" it makes sense.